turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -2565,7 +2565,7 @@ async fn handle_free_var_reference(
         } => {
             let esm_reference = analysis
                 .add_esm_reference_free_var(request.clone(), async || {
-                    Ok(EsmAssetReference::new(
+                    Ok(EsmAssetReference::new_pure(
```

Comment:
add a comment for why this should be pure

im guessing, if we are replacing a free var with an imported symbol, then we really only want the symbol and do not want to expect any side effects

Reply:
will do, but yes:

> do not want to expect any side effects

exactly:

- there is no import if you don't reference `process`
- i.e. it's fine to remove the import again if you referenced `process` but that turned out to be dead code in some later stage of the build pipeline